logbook-api/src/main/java/org/zalando/logbook/attributes/RequestAttributesExtractor.java

```diff
@@ -0,0 +1,24 @@
+package org.zalando.logbook.attributes;
+
+import org.apiguardian.api.API;
+import org.zalando.logbook.HttpRequest;
+
+import javax.annotation.Nonnull;
+
+import static org.apiguardian.api.API.Status.STABLE;
+
+@API(status = STABLE)
+public abstract class RequestAttributesExtractor {
```

Comment:
This should be an interface with just the `extract` method. The `extractOrEmpty` is only needed internally and shouldn't be exposed (or imposed) on extractor implementations.

Reply:
Made it an interface and removed the extra method.